nbextensions/appCell2/widgets/tabs/jobStateListRow.js

```diff
@@ -71,156 +64,64 @@ define([
                 color: color,
                 fontWeight: 'bold'
             },
-            class: icon
-        }, label);
-    }
-
-    function updateRowStatus(ui, jobState, container, clickFunction) {
-        if(!jobState){
-            return;
-        }
-        var selector = '[data-element-job-id="' + jobState.job_id + '"]';
-        var row = container.querySelector(selector);
-        if(row === null){
-            row = document.createElement('tr');
-            row.setAttribute('data-element-job-id', jobState.job_id);
-            if(clickFunction){
-                row.onclick = () => {clickFunction(jobState.job_id)};
-            }
-            container.appendChild(row);
-        }        
-        var jobStatus = jobState ? jobState.job_state : 'Determining Job State...';
-        row.innerHTML = th(jobState.job_id) + niceState(jobStatus);
+        }, [
+            span({
+                class: icon
+            }),
+            ' ' + label
+        ]);
     }
 
     function factory() {
-        var container, ui, listeners = [],
-            jobState = null,
-            runtime = Runtime.make(),
-            listeningForJob = false,
-            jobId;
-
-
-        function startJobUpdates() {
-            if (listeningForJob) {
-                return;
+        var container,
+            ui,
+            name,
+            jobId,
+            clickFunction,
+            isParentJob;
+
+        function updateRowStatus(jobStatus) {
+            jobStatus = jobStatus ? jobStatus : 'Job still pending.';
+            var jobIdDiv = '';
+            if (jobId) {
+                jobIdDiv = div({'style': 'font-size:8pt; color:gray'}, [jobId]);
             }
-            runtime.bus().emit('request-job-update', {
```

Comment:
since request-job-update is removed, how does the state get updated now?

Reply:
The jobStateList now gets its update that contains all child job information. It pushes the child job info into each jobStateListRow widget. Before, each widget was individually requesting and waiting on results from the kernel for its job, so it was bogging things down. Now, on the usual update loop, the owner widget just updates the state for each child job.

Reply:
ops skimmed over that